### PR TITLE
IJ-143 Pagination Refactor

### PR DESCRIPTION
**Pull request by IeuanSkinner**
I've refactored the pagination so that this functionality is integrated into a controller via composition rather than inheritance which has cleaned up a lot of the repeating code.

### CODE REVIEW

File: app/controllers/concerns/pagination.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# app/controllers/concerns/pagination.rb
+module Pagination
+  extend ActiveSupport::Concern
+  included do
+    before_action :pagination, only: :index
+  end
+
+  protected
+
+  # rubocop:disable Metrics/AbcSize
```
Comment (phil-6) [inline app/controllers/concerns/pagination.rb:10]: I love this, sometimes rubocop does need to just do one.

File: app/controllers/team_members/users_controller.rb [hunk 2]
```diff
@@ -1,17 +1,15 @@
...
 
     # GET /users
-    def index
-      @resources_per_page = 6
-      @pinned_users = current_team_member.pinned_users.order(:order)
-      @active_users = User.where(last_sign_in_at: (Time.zone.now - 30.days)..Time.zone.now).count
-      @user_count = User.count
-      super
-    end
+    def index; end
```
Comment (phil-6) [inline app/controllers/team_members/users_controller.rb:12]: do we need this?

File: app/controllers/team_members/wellbeing_assessments_controller.rb [hunk 3]
```diff
@@ -16,18 +12,21 @@ class WellbeingAssessmentsController < PaginationController
     # GET /wellbeing_assessments
     # GET /team_members/:team_member_id/wellbeing_assessments
     # GET /users/:user_id/wellbeing_assessments
-    def index
-      @resources_per_page = 6
-      super
-    end
+    def index; end
```
Comment (phil-6) [inline app/controllers/team_members/wellbeing_assessments_controller.rb:15]: I don't think the empty methods do anything

**Review by phil-6 — APPROVED** [review]
This is a really great refactor and a great piece of work to get done is such a short amount of time. Tons of code duplication removed and overall a really great implementation for pagination.
Great work Ieuan!

My only request is that this gets merged after the other open PRs as there are going to be some conflicts!

File: app/controllers/team_members/users_controller.rb (hunk 2 above)
Comment (IeuanSkinner) [reply app/controllers/team_members/users_controller.rb:12]: They're not needed but they're referenced by the action filters so your IDE will mark them unless you suppress that inspection,  do you want them left in or removed given this?

Comment (phil-6) [reply app/controllers/team_members/users_controller.rb:12]: ahh, I see, lets leave them in to keep JetBrains happy. :airplane: :brain: :smile:

**Pull request merged**